### ISSUE TITLE
bcftools merge sensitive to sample order

### ISSUE THREAD

**Issue by johanneskoester** [samtools/bcftools#241]
When merging bcf files with bcftools merge, the result for loci in which one sample has only the reference allele and the other has a SNP is dependent on the order of the samples.

# Example
Created with samtools mpileup from the files `mpileup.1.bam` and `mpileup.2.bam` from the samtools/tests/mpileup directory.

`mpileup.1.bcf` from `mpileup.1.bam`:
```
17	355	.	G	T,<X>	0	.	DP=9;I16=4,4,0,1,308,12048,41,1681,387,20523,60,3600,159,3439,25,625;QS=0.875,0.125,0;SGB=-0.379885;RPB=1;MQB=1;MQSB=0.406082;BQB=1;MQ0F=0	PL	14,0,200,38,203,231
```

`mpileup.2.bcf` from `mpileup.2.bam`:
```
17	355	.	G	<X>	0	.	DP=9;I16=5,4,0,0,319,11479,0,0,455,24651,0,0,166,3802,0,0;QS=1,0;MQSB=0.89338;MQ0F=0	PL	0,27,222
```

Merging mpileup.1.bcf with mpileup.2.bcf seems to produce the expected output.

```
17	355	.	G	T,<X>	0	.	DP=9;I16=4,4,0,1,308,12048,41,1681,387,20523,60,3600,159,3439,25,625;SGB=-0.379885;RPB=1;MQB=1;MQSB=0.406082;BQB=1;MQ0F=0;QS=1,0.125,0	PL	14,0,200,38,203,231	0,.,.,27,.,222
```

Merging mpileup.2.bcf with mpileup.1.bcf yields:

```
17	355	.	G	T,<X>	0	.	DP=9;I16=4,4,0,1,308,12048,41,1681,387,20523,60,3600,159,3439,25,625;SGB=-0.379885;RPB=1;MQB=1;MQSB=0.406082;BQB=1;MQ0F=0;QS=0.875,0.125,0	PL	.	14,0,200,38,203,231
```
Which is obviously incorrect, since we indeed have values for T and <X> in the PL field in mpileup.2.bcf.

**Comment by johanneskoester** [comment]
I apologize. I just noticed that a second record is created above in the second case. So, in fact, the information is not lost, but we have a duplicate entry with one order, and a single entry with the other order. I guess this should not be order dependent either, right?
Using `-m all` enforces a single record for me.

**Comment by mcshane** [comment]
This issue is resolved on @pd3's experimental branch with support for gvcf merging. This will be brought in after 1.3 lands.

**Comment by mcshane** [comment]
This should be resolved now that #455 has been merged.
